src/model/smp.c

```diff
@@ -15,7 +15,12 @@ void migrateTCB(tcb_t *tcb, word_t new_core)
 #ifdef CONFIG_DEBUG_BUILD
     tcbDebugRemove(tcb);
 #endif
-    Arch_migrateTCB(tcb);
+#ifdef CONFIG_HAVE_FPU
+    /* check if thread owns its current core's FPU */
```

Comment:
I'd change the comment to the less awkward: "Check if thread owns the current core's FPU"

Reply:
I've made this a bit more verbose now. However, having loooked deeper into this, I wonder if we should add another check in this function, if the core really changes (userland may do all koing of strange things) . If the core does not change, then don't do any FPU handling (and maybe don't even call this function). It would be a slight optimization

Reply:
Actually, I opened an issue for that already: #901

Reply:
Cool, that's the next step then after this is merged.